src/proto/beats.proto

```diff
@@ -10,22 +11,75 @@ enum Source {
   USER = 2;
 }
 
-message Beat {
-  optional int32 frame_position = 1;
-  optional bool enabled = 2 [ default = true ];
-  optional Source source = 3 [ default = ANALYZER ];
+enum SectionType {
+  INTRO = 0;
+  OUTRO = 1;
+  OTHER = 2;
+}
+
+// Setting grey as the default color
+message Color {
+  optional int32 red = 1 [ default = 100 ];
+  optional int32 green = 2 [ default = 100 ];
+  optional int32 blue = 3 [ default = 100 ];
 }
 
 message Bpm {
-  optional double bpm = 1;
-  optional Source source = 2 [ default = ANALYZER ];
+  optional double bpm = 1 [ default = 0 ];
+  optional Source source = 2;
 }
 
-message BeatMap {
-  repeated Beat beat = 1;
+message BpmMarker {
+  optional int32 beat_index = 1;
+  optional Bpm bpm = 2;
 }
 
-message BeatGrid {
-  optional Bpm bpm = 1;
-  optional Beat first_beat = 2;
+message TimeSignature {
+  optional int32 beats_per_bar = 1 [ default = 4];
+  optional int32 note_value = 2 [ default = 4 ];
 }
+
+message TimeSignatureMarker {
+  optional int32 downbeat_index = 1 [ default = 0 ];
+  optional TimeSignature time_signature = 2;
+}
+
+message PhraseMarker {
+  optional int32 downbeat_index = 1 [ default = 0 ];
+}
+
+message Section {
+  optional string name = 1 [ default = "UNNAMED"];
+  optional SectionType type = 2 [ default = OTHER ];
+  optional Color color = 3;
+}
+
+message SectionMarker {
+  optional int32 downbeat_index = 1 [default = 0];
```

Comment:
redundant default

Reply:
`first_downbeat_index` would be more specific. This terminology is already used for `Beats`, please consider to apply it consistently for all messages that are defining a range, i.e `Section`, `PhraseMarker`.

Reply:
Naming: Why just `Section` but `PhraseMarker` and not `Phrase`?

Reply:
First downbeat specifies an offset in downbeats for starting bar in a track. It's not relevant for sections. Sections only need to know the bar number aka the downbeat index.